lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -135,9 +136,20 @@ def _on_relation_changed(self, event) -> None:
         )
         if int_tls_ca:
             relation_data[INT_TLS_CA_KEY] = int_tls_ca
-
         self.database_provides.update_relation_data(event.relation.id, relation_data)
 
+    def _on_relation_changed(self, event: RelationChangedEvent) -> None:
+        """Handles providing mongos with KeyFile and hosts."""
+        # First we need to ensure that the database requested event has run
+        # otherwise we would write secrets in plain sight.
```

Comment:
```suggestion
        # First we need to ensure that the database requested event has run
        # otherwise we risk the chance of writing secrets in plain sight.
```